src/operator/tensor/cast_storage.cc

```diff
@@ -32,6 +32,39 @@ namespace op {
 DMLC_REGISTER_PARAMETER(CastStorageParam);
 NNVM_REGISTER_OP(cast_storage)
 .describe(R"code(Casts tensor storage type to the new type.
+
+When an NDArray with default storage type is casted to csr or row_sparse storage,
+the result is compact, which means:
+
+- for csr, zero values will not be retained
+- for row_sparse, row slices with zeros will not be retained
```

Comment:
`row slices with zeros` does not sound accurate. Is `row slices of all zeros` better?